### PR TITLE
Fix bug where secrets were not parsed when using old firebase functions sdk versions

### PR DESCRIPTION
**Pull request by taeold**
Partially fixes https://github.com/firebase/firebase-tools/issues/4797

### CODE REVIEW

File: src/deploy/functions/runtimes/node/parseTriggers.ts [hunk 1]
```diff
@@ -140,6 +140,9 @@ export function useStrategy(): Promise<boolean> {
   return Promise.resolve(true);
 }
 
+/**
+ * Parse trigger annotations in sourceDir to generate backed.Build.
```
Comment (TheIronDev) [inline src/deploy/functions/runtimes/node/parseTriggers.ts:144]: nit: `{@link backend.Build}`

+others

File: src/deploy/functions/runtimes/node/parseTriggers.ts [hunk 2]
```diff
@@ -377,12 +389,24 @@ export function addResourcesToBuild(
     "timeout",
     proto.secondsFromDuration
   );
+  if (annotation.secrets) {
+    const secretEnvs: backend.SecretEnvVar[] = [];
+    for (const secret of annotation.secrets) {
```
Comment (TheIronDev) [inline src/deploy/functions/runtimes/node/parseTriggers.ts:394]: nit: Why not:

```
const secretEnvs: backend.SecretEnvVar[] = annotation.secrets.map((secret) => ({
        secret,
        projectId,
        key: secret,
      }));
```

**Review by TheIronDev — APPROVED** [review]
LGTM,
Can you also write a test for this?

File: src/deploy/functions/runtimes/node/parseTriggers.ts (hunk 1 above)
Comment (taeold) [reply src/deploy/functions/runtimes/node/parseTriggers.ts:144]: We don't use api-documenter for CLI 🤔 Do you think that annotation is useful regardless?

File: src/deploy/functions/runtimes/node/parseTriggers.ts (hunk 2 above)
Comment (taeold) [reply src/deploy/functions/runtimes/node/parseTriggers.ts]: ctrl+c/ctrl+v 😛 (see few lines below). I think this one -changed.

**Review by colerogers — APPROVED** [review]

**Pull request merged**